packages/core/src/utilities/renderToCanvasGPU.ts

```diff
@@ -0,0 +1,120 @@
+import getOrCreateCanvas from '../RenderingEngine/helpers/getOrCreateCanvas';
+import { ViewportType, Events } from '../enums';
+import StackViewport from '../RenderingEngine/StackViewport';
+import { IImage } from '../types';
+import { getRenderingEngine } from '../RenderingEngine/getRenderingEngine';
+import RenderingEngine from '../RenderingEngine';
+
+/**
+ * Renders an cornerstone image to a Canvas. This method will handle creation
+ * of a temporary enabledElement, setting the imageId, and rendering the image via
+ * a StackViewport, copying the canvas drawing to the given canvas Element, and
+ * disabling the created temporary element. SuppressEvents argument is used to
+ * prevent events from firing during the render process (e.g. during a series
+ * of renders to a thumbnail image).
+ *
+ * @example
+ * ```
+ * const canvas = document.getElementById('myCanvas')
+ *
+ * renderToCanvas(canvas, image)
+ * ```
+ * @param canvas - Canvas element to render to
+ * @param image - The image to render
+ * @param modality - [Default = undefined] The modality of the image
+ * @returns - A promise that resolves when the image has been rendered with the imageId
+ */
+export default function renderToCanvasGPU(
+  canvas: HTMLCanvasElement,
+  image: IImage,
+  modality = undefined,
+  renderingEngineId = '_thumbnails'
+): Promise<string> {
+  if (!canvas || !(canvas instanceof HTMLCanvasElement)) {
+    throw new Error('canvas element is required');
+  }
+
+  const imageIdToPrint = image.imageId;
+  const viewportId = `thumbnailViewport-${imageIdToPrint}`;
```

Comment:
Why this viewport id? Why not something more generic like `renderGPUViewport-${imageIdToPring}`? Pardon me if this is a stupid question.